WordPress/gutenberg
# Lodash: Refactor away from `_.keys()`

## What?
Lodash's `keys` is used only a few times in the entire codebase. This PR aims to remove that usage.

## Why?
Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?
Removing `keys` is straightforward in favor of a simple `Object.keys()` replacement. However, we need to also ensure we're not calling it on non-objects (`null` and `undefined` specifically, since on those it will throw an error).

## Testing Instructions
* Verify all tests pass.
* Verify `npm run api-docs:blocks` runs and completes without problems.
* Verify `npm run api-docs:theme-ref` runs and completes without problems.
